utils/led.c

```diff
@@ -27,17 +27,21 @@ volatile uint32_t g_gpio_pattern = LEDS_MASK;
 static uint8_t num_of_blinks, count;
 static uint32_t timer2_load_value = MSS_SYS_APB_0_CLK_FREQ;
 
+/* Local function prototypes */
+void led_set_high(void);
+void led_conf_tim2_periodic(void);
+
+
 /**
- * @brief Blink function for LED at GPIO0
+ * @brief Blink function for LED at GPIO0 called upon MSS power on reset
  *
- * This function is called to make LED at GPIO0 (configured as output) blink
- * a certain number of times defined by the enumeration #led_error_type_t present in
- * led_error.h file.
- *
- * @param blinks[in]      Number of blinks for LED
+ * This function is called is called in main() function at the start to indicate power on
+ * reset of MSS. It shall blink @b LED_RESET number of times (defined in led_error.h).
+ * It configures the GPIO0 in OUTPUT_MODE such that this GPIO0 configuration need not be
+ * changed afterwards before calling led_custom_blink().
  *
  */
-void led_blink(led_error_type_t blinks)
+void led_blink_reset(void)
```

Comment:
My apologies if I was unclear in my review yesterday... I actually very much liked the way your function interface looked like for the LED! A single `led_blink()` function with a param. for the number of times the LED should blink is very understandable! I think the new interface is a bit less intuitive.

My suggestion yesterday was to rename the `enum` members to `LED_BLINK_RESET`, `LED_BLINK_...`, not the functions...

Reply:
No worries! As discussed, `led.c` file will only have two exported functions (in `led.h)` - `led_init()` and `led_tim2_start()`. As for enum fields I will rename them.

Reply:
Thanks!

Reply:
Yeah. It is fixed now :)